File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/api/dto/LoginResponse.java
package br.ufpr.sept.so2.modules.iam.api.dto;

import br.ufpr.sept.so2.modules.iam.application.LoginResult;

public record LoginResponse(
        String accessToken,
        boolean mustChangePassword,
        long expiresIn,
        String tokenType
) {

    public static LoginResponse from(LoginResult result) {
        return new LoginResponse(result.accessToken(), result.mustChangePassword(), result.expiresIn(), "Bearer");
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/api/dto/LoginResponse.kt
package br.ufpr.sept.so2.modules.iam.api.dto

import br.ufpr.sept.so2.modules.iam.application.LoginResult
import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class LoginResponse(
    val accessToken: String,
    val mustChangePassword: Boolean,
    val expiresIn: Long,
    val tokenType: String,
    val refreshToken: String? = null,
) {
    companion object {
        fun from(result: LoginResult, includeRefreshToken: Boolean = false): LoginResponse =
            LoginResponse(
                result.accessToken,
                result.mustChangePassword,
                result.expiresIn,
                "Bearer",
                if (includeRefreshToken) result.refreshToken else null,
            )
    }
}
